### PR TITLE
Add backup command

### PR DESCRIPTION
**Pull request by amalpaul54111**
Added minimum backup commands based on @lexsulzer 's https://github.com/dataspects/mediawiki-cli/tree/canasta/manage-snapshots-canasta

1. init
2. view
3. take-snapshot
4. restore

### CODE REVIEW

File: cmd/restic/check.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	}
+	checkCmd.Flags().StringVarP(&tag, "tag", "t", "", "Restic snapshot ID (required)")
+	return checkCmd
+}
+
+func check() {
+	envPath := instance.Path + "/.env"
+	EnvVariables := canasta.GetEnvVariable(envPath)
+
+	output := execute.Run(instance.Path, "sudo", "docker", "run", "--rm", "-i", "--env-file", envPath, "restic/restic", "-r", "s3:"+EnvVariables["AWS_S3_API"]+"/"+EnvVariables["AWS_S3_BUCKET"], "check")
```
Comment (jeffw16) [inline cmd/restic/check.go]: fix spacing between `+`

also, this seems repetitive to me. Every file will do execute.Run docker run restic etc., the only thing that is different seems to be the last parameter

File: cmd/restic/diff.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	diffCmd.MarkFlagRequired("tag1")
+	diffCmd.MarkFlagRequired("tag2")
+	return diffCmd
+}
+
+func diff() {
+	envPath := instance.Path + "/.env"
+	EnvVariables := canasta.GetEnvVariable(envPath)
+
+	output := execute.Run(instance.Path, "sudo", "docker", "run", "--rm", "-i", "--env-file", envPath, "restic/restic", "-r", "s3:"+EnvVariables["AWS_S3_API"]+"/"+EnvVariables["AWS_S3_BUCKET"], "diff", tag1, tag2)
```
Comment (jeffw16) [inline cmd/restic/diff.go]: fix spacing between `+`

File: cmd/restic/forgetSnapshots.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+	forgetSnapshotCmd.Flags().StringVarP(&tag, "tag", "t", "", "Restic snapshot ID (required)")
+	return forgetSnapshotCmd
+}
+
+func forgetSnapshot() {
+	envPath := instance.Path + "/.env"
+	EnvVariables := canasta.GetEnvVariable(envPath)
+
+	output := execute.Run(instance.Path, "sudo", "docker", "run", "--rm", "-i", "--env-file", envPath, "restic/restic", "-r", "s3:"+EnvVariables["AWS_S3_API"]+"/"+EnvVariables["AWS_S3_BUCKET"], "forget", tag)
```
Comment (jeffw16) [inline cmd/restic/forgetSnapshots.go]: fix

File: cmd/restic/init.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	}
+	return initCmd
+}
+
+func initRestic() {
+	fmt.Println("Initializing Restic repo in S3")
+	envPath := instance.Path + "/.env"
+	EnvVariables := canasta.GetEnvVariable(envPath)
+
+	execute.Run(instance.Path, "sudo", "docker", "run", "--rm", "-i", "--env-file", envPath, "restic/restic", "-r", "s3:"+EnvVariables["AWS_S3_API"]+"/"+EnvVariables["AWS_S3_BUCKET"], "init")
```
Comment (jeffw16) [inline cmd/restic/init.go]: fix

File: cmd/restic/listFiles.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	}
+	listFilesCmd.Flags().StringVarP(&tag, "tag", "t", "", "Restic snapshot ID (required)")
+	return listFilesCmd
+}
+
+func listFiles() {
+	envPath := instance.Path + "/.env"
+	EnvVariables := canasta.GetEnvVariable(envPath)
+
+	output := execute.Run(instance.Path, "sudo", "docker", "run", "--rm", "-i", "--env-file", envPath, "restic/restic", "-r", "s3:"+EnvVariables["AWS_S3_API"]+"/"+EnvVariables["AWS_S3_BUCKET"], "ls", tag)
```
Comment (jeffw16) [inline cmd/restic/listFiles.go]: fix

File: cmd/restic/unlock.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+		},
+	}
+	return unlockCmd
+}
+
+func unlock() {
+	envPath := instance.Path + "/.env"
+	EnvVariables := canasta.GetEnvVariable(envPath)
+
+	output := execute.Run(instance.Path, "sudo", "docker", "run", "--rm", "-i", "--env-file", envPath, "restic/restic", "-r", "s3:"+EnvVariables["AWS_S3_API"]+"/"+EnvVariables["AWS_S3_BUCKET"], "unlock")
```
Comment (jeffw16) [inline cmd/restic/unlock.go]: see above

File: cmd/restic/viewSnapshots.go [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+		},
+	}
+	return initCmd
+}
+
+func viewSnapshots() {
+	envPath := instance.Path + "/.env"
+	EnvVariables := canasta.GetEnvVariable(envPath)
+
+	output := execute.Run(instance.Path, "sudo", "docker", "run", "--rm", "-i", "--env-file", envPath, "restic/restic", "-r", "s3:"+EnvVariables["AWS_S3_API"]+"/"+EnvVariables["AWS_S3_BUCKET"], "snapshots")
```
Comment (jeffw16) [inline cmd/restic/viewSnapshots.go]: see above

**Review by jeffw16 — CHANGES_REQUESTED** [review]

File: cmd/restic/check.go (hunk 1 above)
Comment (amalpaul54111) [reply cmd/restic/check.go]: About the spacing between ` + `. 
Go provides a tool named `gofmt` and VS Code uses the same to format the code. And it prefers to remove the space between the `+` always.

Yeah, I will also fix the repetitive usage of the sudo docker run.

**Comment by yaronkoren** [comment]
Yes, indeed - it looks like people have complained about gofmt's strange spacing for a long time, but it's what the developers want:

https://github.com/golang/go/issues/12720

Given that, I think we should stick with gofmt's output.

**Review by jeffw16 — APPROVED** [review]

**Pull request merged**